discovery/watcher.go

```diff
@@ -285,6 +287,7 @@ func (w *Watcher) nextServer(addrs *addrSet) (*addrSet, error) {
 	}
 	if len(healthy) == 0 {
 		// No healthy servers. Re-run discovery.
+		metrics.SetGauge([]string{"consul_connected"}, 0)
```

Comment:
Instead of here, I think we should set the `consul_connected` guage to zero (1) at the top of `nextServer` and (2) when `nextServer` returns. Basically, `nextServer` is called when we need to find a server to connect to. And `nextServer` blocks (in the `watch()` function) until we disconnect for some reason, which is the only case it returns.

So you could set this to zero in `run()` both before and after `nextServer` is called.

Reply:
That's super helpful. Wasn't exactly sure where the best place was to put this :). Thank you!

Reply:
So to clarify 0 before/after `nextServer` and set to 1 after the actual connected event at L327 in watcher.go?